### PR TITLE
Feature/summernote

### PR DESCRIPTION
**Pull request by tsteinruecken**
Replacement for #8 - support for summernote editor.

### CODE REVIEW

File: bones/textBone.py [hunk 1]
```diff
@@ -46,71 +50,156 @@ def handle_data(self, data):
...
-	def cleanup(self): #FIXME: vertauschte tags
+	def cleanup(self):  # FIXME: vertauschte tags
 		""" Append missing closing tags """
+		self.flushCache()
 		for tag in self.openTagsList:
 			endTag = '</%s>' % tag
 			self.result += endTag
 
-	def santinize( self, instr ):
+	def santinize(self, instr):
```
Comment (phorward) [inline bones/textBone.py]: Shouldn't this be called "sanitize"?

File: bones/textBone.py [hunk 2]
```diff
@@ -46,71 +50,156 @@ def handle_data(self, data):
...
+				# Close all currently open Tags until we reach the current one. If no one is found,
+				# we just close everything and ignore the tag that should have been closed
+				for endTag in self.openTagsList[:]:
+					self.result += "</%s>" % endTag
+					self.openTagsList.remove(endTag)
+					if endTag == tag:
+						break
 
-	def cleanup(self): #FIXME: vertauschte tags
+	def cleanup(self):  # FIXME: vertauschte tags
```
Comment (phorward) [inline bones/textBone.py]: what's meant with this fixme?

File: bones/textBone.py [hunk 3]
```diff
@@ -46,71 +50,156 @@ def handle_data(self, data):
...
+			cacheTagStart = '<' + tag
 			isBlankTarget = False
+			styles = None
+			classes = None
 			for k, v in attrs:
-				if not tag in self.validHtml["validAttrs"] or not k in self.validHtml["validAttrs"][ tag ]:
+				k = k.strip()
+				v = v.strip()
+				if any([c in k for c in filterChars]) or any([c in v for c in filterChars]):
+					if (k == "title" or k == "href") and not any([c in v for c in "\"'\\\0\r\n"]):
```
Comment (sveneberth) [inline bones/textBone.py]: I think we should allow also `@` and `()` in `alt`-attributes.

File: bones/textBone.py (hunk 2 above)
Comment (sveneberth) [reply bones/textBone.py]: @tsteinruecken What about this point?

**Pull request merged**